mono/metadata/icall.c

```diff
@@ -279,8 +279,100 @@ set_invalid_cast (MonoError *error, MonoClass *src_class, MonoClass *dst_class)
 	mono_error_set_invalid_cast (error);
 }
 
+#if ENABLE_NETCORE
+void
+ves_icall_System_Array_SetValueRelaxedImpl (MonoArrayHandle arr, MonoObjectHandle value, guint32 pos, MonoError *error)
+{
+	array_set_value_impl (arr, value, pos, FALSE, FALSE, error);
+}
+
+// Copied from CoreCLR: https://github.com/dotnet/coreclr/blob/d3e39bc2f81e3dbf9e4b96347f62b49d8700336c/src/vm/invokeutil.cpp#L33
+#define PRIMITIVE_TABLE_SIZE  MONO_TYPE_STRING
```

Comment:
This `#define` isn't really needed, you can just say `static const guint32 primitive_conversions [ ] = { … }` without an explicit size, unless you know this is the size and are double checking yourself, that you didn't put in extra entries (too few entries won't be caught, just zero-padded).

Reply:
@jaykrell `PRIMITIVE_TABLE_SIZE` is used inside `can_primitive_widen` to catch types that can not be widen or require special case ((U)IntPtr)